### PR TITLE
[stdlib] Make KeyValuePairs conditionally conform to Equatable and Hashable

### PR DESCRIPTION
**Pull request by calda**
This PR makes the standard library type `KeyValuePairs` conditionally conform to `Equatable` and `Hashable`:

```swift
extension KeyValuePairs: Equatable where Key: Equatable, Value: Equatable { /* ... */ }
extension KeyValuePairs: Hashable where Key: Hashable, Value: Hashable { /* ... */ }
```

Does this seem like it could be added without a formal Swift Evolution review cycle? It affects the public API of a type, but it also seems like the obviously correct behavior.

[This SE thread](https://forums.swift.org/t/keyvaluepairs-conditional-conformance-to-equatable-hashable/32730) noted that this change would require support for [Backwards-deployable Conformances](https://forums.swift.org/t/backwards-deployable-conformances/29876). Do we have any mechanisms for that at this time? Any guidance would be appreciated!

### CODE REVIEW

File: stdlib/public/core/KeyValuePairs.swift [hunk 1]
```diff
@@ -125,6 +125,56 @@ extension KeyValuePairs: RandomAccessCollection {
...
+    /// contain the same elements in the same order.
+    ///
+    /// You can use the equal-to operator (`==`) to compare any two `KeyValuePairs` instances
+    /// that store the same, `Equatable`-conforming `Key` and `Value` types.
+    ///
+    /// - Parameters:
+    ///   - lhs: An `KeyValuePairs` instance to compare.
+    ///   - rhs: Another `KeyValuePairs` instance to compare.
+    @inlinable
+    public static func ==(lhs: KeyValuePairs<Key, Value>, rhs: KeyValuePairs<Key, Value>) -> Bool {
```
Comment (calda) [inline stdlib/public/core/KeyValuePairs.swift]: Modeled after the existing [`Array.==(lhs:rhs:)` implementation](https://github.com/apple/swift/blob/516ad28c504d835e4c189930020016a96d72bd06/stdlib/public/core/Array.swift#L1730)

**Comment by theblixguy** [comment]
I don’t think there is a way yet (which is why some evolution [proposals](https://github.com/apple/swift-evolution/pull/973) that depend on this feature are currently blocked).

**Comment by airspeedswift** [comment]
Thanks for contributing – but as you note, this cannot be done until we have back-deployment of conformances. Closing this PR until that is available.

**Comment by airspeedswift** [comment]
> Do we have any mechanisms for that at this time?

I'm afraid not. We'll have to wait for that feature before proceeding.

**Pull request closed without merging**